nntrainer/layers/mol_attention_layer.cpp

```diff
@@ -72,34 +137,55 @@ void MoLAttentionLayer::forwarding(RunLayerContext &context, bool training) {
   Tensor &state = context.getInput(wt_idx[AttentionParams::state]);
 
   Tensor &output = context.getOutput(SINGLE_INOUT_IDX);
-  Tensor &mlp_w = context.getWeight(wt_idx[AttentionParams::mlp_w]);
-  Tensor &mlp_proj_w = context.getWeight(wt_idx[AttentionParams::mlp_proj_w]);
+  Tensor &fc_w = context.getWeight(wt_idx[AttentionParams::fc_w]);
+  Tensor &fc_bias = context.getWeight(wt_idx[AttentionParams::fc_bias]);
+  Tensor &fc_proj_w = context.getWeight(wt_idx[AttentionParams::fc_proj_w]);
+  Tensor &fc_out = context.getTensor(wt_idx[AttentionParams::fc_out]);
+  Tensor &fc_tanh = context.getTensor(wt_idx[AttentionParams::fc_tanh]);
+  Tensor &fc_proj_out = context.getTensor(wt_idx[AttentionParams::fc_proj_out]);
+  Tensor &scores = context.getTensor(wt_idx[AttentionParams::scores]);
+  Tensor &prob = context.getTensor(wt_idx[AttentionParams::prob]);
+  Tensor &prob_left = context.getTensor(wt_idx[AttentionParams::prob_left]);
+  Tensor &prob_right = context.getTensor(wt_idx[AttentionParams::prob_right]);
+  Tensor &u_neg_div = context.getTensor(wt_idx[AttentionParams::u_neg_div]);
+  Tensor &u_pos_div = context.getTensor(wt_idx[AttentionParams::u_pos_div]);
 
   const TensorDim &input_dim = query.getDim();
   unsigned int batch = input_dim.batch();
   auto mol_k = std::get<props::MoL_K>(mol_props).get();
 
-  Tensor mlp_out = query.dot(mlp_w);
+  /** reset helper state */
+  helper_exec = false;
+
+  fc_out = query.dot(fc_w);
```

Comment:
query.dot(fc_w, fc_out) to maintain reference?

Reply:
this is done in the last commit.